sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataListTest.java

```diff
@@ -0,0 +1,655 @@
+import com.microsoft.appcenter.data.models.Page;
+import com.microsoft.appcenter.data.models.PaginatedDocuments;
+import com.microsoft.appcenter.data.models.ReadOptions;
+import com.microsoft.appcenter.data.models.TokenResult;
+import com.microsoft.appcenter.data.models.WriteOptions;
+import com.microsoft.appcenter.http.HttpClient;
+import com.microsoft.appcenter.http.HttpException;
+import com.microsoft.appcenter.http.ServiceCall;
+import com.microsoft.appcenter.http.ServiceCallback;
+import com.microsoft.appcenter.utils.async.AppCenterFuture;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+
+import org.hamcrest.CoreMatchers;
+import org.json.JSONException;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+import static com.microsoft.appcenter.data.Constants.PENDING_OPERATION_CREATE_VALUE;
+import static com.microsoft.appcenter.data.Constants.PENDING_OPERATION_DELETE_VALUE;
+import static com.microsoft.appcenter.data.Constants.PENDING_OPERATION_REPLACE_VALUE;
+import static com.microsoft.appcenter.data.Constants.PREFERENCE_PARTITION_PREFIX;
+import static com.microsoft.appcenter.data.DefaultPartitions.APP_DOCUMENTS;
+import static com.microsoft.appcenter.data.DefaultPartitions.USER_DOCUMENTS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyMapOf;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.endsWith;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.startsWith;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.verifyNoMoreInteractions;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PrepareForTest({
+        CosmosDb.class,
+        TokenExchange.class,
+        TokenManager.class
+})
+public class DataListTest extends AbstractDataTest {
+
+    @Captor
+    private ArgumentCaptor<DocumentWrapper<TestDocument>> mTestDocumentWrapperCaptor;
+
+    private void mockSignOut() {
+        Mockito.when(mAuthTokenContext.getAccountId()).thenReturn(null);
+    }
+
+    @Before
+    public void setUpAuth() {
```

Comment:
Isn't that duplicated also?

Reply:
Yes, it is somewhat duplicating across files. But because there are so many subclasses off `AbstractDataTest`, there are a few other tests in some other files that assume that auth context is not set up. So if I moved this one under a `@Before` in the parent class, other tests would need to be adjusted.